import-service/handler.ts

```diff
@@ -0,0 +1,64 @@
+import { APIGatewayProxyHandler, S3Handler } from 'aws-lambda'
+
+import S3 from 'aws-sdk/clients/s3'
+import csvParser from 'csv-parser'
+
+import 'source-map-support/register'
+
+const BUCKET = 'imported-csv'
+
+export const catalogUpload: APIGatewayProxyHandler = (event) => {
+  const catalogName = event.queryStringParameters.name
+  const catalogPath = `uploaded/${catalogName}`
+  const s3 = new S3({ region: 'eu-west-1' })
+  const params = { 
+    Bucket: BUCKET,
+    Key: catalogPath,
+    Expires: 60,
+    ContentType: 'text/csv'
+  }
+
+  return new Promise((resolve, reject)=>{
+    s3.getSignedUrl('putObject', params, (err, url)=>{
+      if(err) return reject(err)
+
+      resolve({
+        statusCode: 200,
+        headers: { 'Access-Control-Allow-Origin': '*' },
+        body: url
+      })
+    })
+  })
```

Comment:
In case of error it's better to respond with valid response structure.
E.g.:
```
reject({
      statusCode: 500,
      body: JSON.stringify({
        message: err.message,
      }),
})
```
Also, as an alternative to `new Promse((resolve, reject) => .....)` you can use just `getSignedUrlPromise`
```
await s3.getSignedUrlPromise("putObject", params);
```